golem-cli/src/wasm_rpc_stubgen/stub.rs

```diff
@@ -668,48 +659,116 @@ struct WorldItemsByType<'a> {
     pub interfaces: Vec<(String, &'a Interface)>,
 }
 
-#[derive(Debug, Clone)]
-pub struct InterfaceStub {
+pub struct StubbedWorld {
+    pub name: String,
+    pub functions: Vec<FunctionStub>,
+    pub used_types: Vec<TypeId>,
+}
+
+pub struct StubbedInterface {
+    pub name: String,
+    pub functions: Vec<FunctionStub>,
+    // name of the interface in case its not an inline interface
+    pub interface_name: Option<String>,
+    pub used_types: Vec<TypeId>,
+}
+
+pub struct StubbedResource {
     pub name: String,
     pub constructor_params: Option<Vec<FunctionParamStub>>,
     pub functions: Vec<FunctionStub>,
     pub static_functions: Vec<FunctionStub>,
-    pub used_types: Vec<TypeId>,
-    pub global: bool,
     pub owner_interface: Option<String>,
 }
 
-impl InterfaceStub {
-    // The returned bool is true if the function is static
-    pub fn all_functions(&self) -> impl Iterator<Item = (&FunctionStub, bool)> {
-        self.static_functions
-            .iter()
-            .map(|f| (f, true))
-            .chain(self.functions.iter().map(|f| (f, false)))
-    }
+pub enum StubbedEntity {
```

Comment:
This refactoring was mostly to convince myself that I'm dealing with types properly here.
If desired I can revert this part and keep just the bugfix

Reply:
The "stubbed world" naming was a bit confusing to me, as it sounds like as the stub for the whole world (when the Stub itself is that).  If i understand correctly, it means "StubbedWorldFunctions".